packages/adapters/examples/segmentationExport/index.ts

```diff
@@ -0,0 +1,328 @@
+);
+
+const { Cornerstone3D } = adaptersSEG;
+
+const {
+    SegmentationDisplayTool,
+    StackScrollMouseWheelTool,
+    ToolGroupManager,
+    Enums: csToolsEnums,
+    segmentation
+} = cornerstoneTools;
+
+const { ViewportType } = Enums;
+
+// Define a unique id for the volume
+const volumeName = "CT_VOLUME_ID"; // Id of the volume less loader prefix
+const volumeLoaderScheme = "cornerstoneStreamingImageVolume"; // Loader id which defines which volume loader to use
+const volumeId = `${volumeLoaderScheme}:${volumeName}`; // VolumeId with loader id + volume id
+const segmentationId = "MY_SEGMENTATION_ID";
+const toolGroupId = "MY_TOOLGROUP_ID";
+// Create the viewports
+const viewportId1 = "CT_AXIAL";
+const viewportId2 = "CT_SAGITTAL";
+const viewportId3 = "CT_CORONAL";
+
+// ======== Set up page ======== //
+setTitleAndDescription(
+    "DICOM SEG Export",
+    "Here we demonstrate how to export a DICOM SEG from a Cornerstone3D volume."
+);
+
+const size = "500px";
+const content = document.getElementById("content");
+const viewportGrid = document.createElement("div");
+
+viewportGrid.style.display = "flex";
+viewportGrid.style.display = "flex";
+viewportGrid.style.flexDirection = "row";
+
+const element1 = document.createElement("div");
+const element2 = document.createElement("div");
+const element3 = document.createElement("div");
+element1.style.width = size;
+element1.style.height = size;
+element2.style.width = size;
+element2.style.height = size;
+element3.style.width = size;
+element3.style.height = size;
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+viewportGrid.appendChild(element3);
+
+content.appendChild(viewportGrid);
+let renderingEngine;
+let segmentationVolume;
+let segmentationRepresentationUID;
+
+function generateMockMetadata(segmentIndex, color) {
+    const RecommendedDisplayCIELabValue = dcmjs.data.Colors.rgb2DICOMLAB(
+        color.slice(0, 3).map(value => value / 255)
+    ).map(value => Math.round(value));
+
+    return {
+        SegmentedPropertyCategoryCodeSequence: {
+            CodeValue: "T-D0050",
+            CodingSchemeDesignator: "SRT",
+            CodeMeaning: "Tissue"
+        },
+        SegmentNumber: segmentIndex.toString(),
+        SegmentLabel: "Tissue " + segmentIndex.toString(),
+        SegmentAlgorithmType: "SEMIAUTOMATIC",
+        SegmentAlgorithmName: "Slicer Prototype",
+        RecommendedDisplayCIELabValue,
+        SegmentedPropertyTypeCodeSequence: {
+            CodeValue: "T-D0050",
+            CodingSchemeDesignator: "SRT",
+            CodeMeaning: "Tissue"
+        }
+    };
+}
+
+// ============================= //
+
+addButtonToToolbar({
+    title: "Create SEG",
+    onClick: async () => {
+        const volume = cache.getVolume(volumeId);
+
+        const volumeImages = volume.convertToCornerstoneImages();
+        const imagePromises = volumeImages.map(image => image.promise);
+
+        await Promise.all(imagePromises).then(images => {
```

Comment:
Is PixelData needed to generate the segmentation, or metadata would be enough?

Reply:
I believe dcmjs only need metadata to create a derived multiframe SEG from the reference